app/styles/_data-toolbar.less

```diff
@@ -3,45 +3,38 @@
 ---------------------------------------------------------------------------- */
 
 .data-toolbar {
-  .flex(@columns: 1 1 0%);
-  .flex-direction(@direction: column);
-  .flex-display(@display: flex);
-  .flex-wrap(@wrap: wrap);
   padding: 5px 0;
   @media (min-width: @screen-sm-min) {
-    .flex-direction(@direction: row);
+    .flex-display(@display: flex);
   }
   // Set min-widths to prevent clipping when option lengths are known
   &.other-resources-toolbar .data-toolbar-dropdown {
-    min-width: 250px;
+    min-width: 210px;
   }
   .checkbox {
     margin-bottom: 0;
     margin-top: 10px;
-    width: 100%;
-    @media (min-width: 900px) {
+    @media (min-width: @screen-sm-min) {
       .flex(@columns: 1 1 0%);
+      margin-left: 20px;
       margin-top: 3px;
       text-align: right;
-      width: auto;
     }
   }
-  .data-toolbar-dropdown {
-    min-width: 160px;
-  }
   .data-toolbar-filter {
-    min-width: 75px;
     @media (min-width: @screen-sm-min) {
-      // Set flex rules so that filter expands in IE11
-      .flex-display(@display: flex);
-      .flex(@columns: 1 1 auto);
+      .flex(@columns: 1 1 0%);
```

Comment:
@rhamilto Can you double check Safari if you haven't? I've seen problems with overlapping elements with flex basis 0%